teps/0047-pipeline-task-display-name.md

```diff
@@ -0,0 +1,181 @@
+is the machine readable `Pipeline.spec.tasks.*.name` field.
+
+Compare this to other tools in this space, Jenkins, GitLab CI, GitHub Actions, etc.
+Their renderings of their workflows are meant to be human consumable. Their task names have
+the option to be rendered in a human consumable format with use of any characters and fully
+rendered for context.
+
+### Goals
+
+* Add a way to specify an optional display name for `Pipeline.spec.tasks.*.` items that allows
+  any text and if provided will be useable by UIs rendering Tekton Piplines rather then the
+  machine readable `Pipeline.spec.tasks.*.name` field.
+* Add a way to specify a long form description for `Pipeline.spec.tasks.*.` items that allows for a
+  long form description of a task in the context of a `Pipeline` to provide additional
+  human context to the intent of a task in a Pipeline.
+* Add a way to specify an optional display name for `Pipeline` objects that allows any text
+  and if provided will be useable for UIs rendering Tekton `Pipeline` rather then the
+  machine readable `Pipeline.name` field.
+### Non-Goals
+
+To be determined.
+
+### Use Cases (optional)
+
+* Pipeline writers can optionally specify a display name for any/all tasks in a `Pipeline`.
+* Pipeline writers can optionally specify a description for any/all tasks in a `Pipeline`.
+* Pipeline writers can optionally specific a display name for a `Pipeline`.
+* Pipeline writers can optionally specify a description for a `Pipeline`.
+## Requirements
+
+* New display name filed would be optional and accept any unicode character.
+* New description field would be optional and accept any unicode character.
+
+## Proposal
+
+See goals and alternatives.
+### Notes/Caveats (optional)
+
+None.
+### Risks and Mitigations
+
+1. there are limits on the lengths of strings Kubernetes will store which may need some handling
+of some sort
+
+2. Tools that render Tekton pipelines will need to be updated to take advantage of the new
+fields once they are available.
+### User Experience (optional)
+
+None.
+
+### Performance (optional)
+
+Non predicted.
+
+## Design Details
+
+Suggested new parameters would be:
+* `Pipeline.spec.tasks.*.displayName`
+* `Pipeline.spec.tasks.*.description`
+* `Pipeline.displayName`
+* `Pipeline.description`
+
+## Test Plan
+
+To be determined.
+
+## Design Evaluation
+None.
+
+## Drawbacks
+None.
+
+## Alternatives
+
+### Possible solution 1
+Add new fields to spec.
+```
+kind: Pipeline
+metadata:
+  name: our-awesome-pipeline
+spec:
+  displayName: Our Awesome Pipeline
+  description: |
+    This is our awesome pipeline that does amazing things.
+  tasks:
+  - name: setup-pgp-keys
+    displayName: SETUP: PGP Keys
+    description: |
+      Sets up the PGP keys required to decrypt encrypted configuration.
+    taskRef:
+      kind: ClusterTask
+      name: import-pgp-keys
+```
+
+### Possible solution 2
+Add new labels to metadata.
```

Comment:
Also suggest: "Add metadata field to PipelineTasks."

We currently don't have `metadata` field in the `pipeline.spec.tasks.*` so it would be good to note that here too.

Reply:
updated example description a bit.